File: 2833-furthest-point-from-origin/solution.py
class Solution:
    def furthestDistanceFromOrigin(self, moves: str) -> int:
        """
        Calculates the furthest point from the origin.
        
        The strategy is to find the net difference between fixed 'L' and 'R' moves
        and add the total number of flexible '_' moves to that magnitude.
        """
        # Count occurrences of each character
        count_L = moves.count('L')
        count_R = moves.count('R')
        count_underscore = moves.count('_')
        
        # Max distance = magnitude of fixed moves + all flexible moves
        return abs(count_L - count_R) + count_underscore

File: 2833-furthest-point-from-origin/solution.cpp
#include <string>
#include <cmath>
#include <algorithm>

class Solution {
public:
    /**
     * Calculates the furthest point from the origin by counting the frequency
     * of each move type and greedily assigning the underscores.
     */
    int furthestDistanceFromOrigin(std::string moves) {
        int countL = 0;
        int countR = 0;
        int countUnderscore = 0;

        for (char move : moves) {
            if (move == 'L') {
                countL++;
            } else if (move == 'R') {
                countR++;
            } else {
                countUnderscore++;
            }
        }

        // The logic: abs(net_fixed_distance) + total_flexible_moves
        return std::abs(countL - countR) + countUnderscore;
    }
};